app/src/main/java/org/mifos/selfserviceapp/ui/fragments/LoanApplicationFragment.java

```diff
@@ -119,31 +120,38 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
-        loansPayload.setNumberOfRepayments(loanTemplate.getNumberOfRepayments());
-        loansPayload.setRepaymentEvery(loanTemplate.getRepaymentEvery());
-        loansPayload.setRepaymentFrequencyType(loanTemplate.getInterestRateFrequencyType().getId());
-        loansPayload.setInterestRatePerPeriod(loanTemplate.getInterestRatePerPeriod());
-        loansPayload.setExpectedDisbursementDate(disbursementDate);
-        loansPayload.setSubmittedOnDate(submittedDate);
-
-        loansPayload.setTransactionProcessingStrategyId(
-                loanTemplate.getTransactionProcessingStrategyId());
-        loansPayload.setAmortizationType(loanTemplate.getAmortizationType().getId());
-        loansPayload.setTransactionProcessingStrategyId(
-                loanTemplate.getTransactionProcessingStrategyId());
-        loansPayload.setInterestCalculationPeriodType(
-                loanTemplate.getInterestCalculationPeriodType().getId());
-        loansPayload.setInterestType(loanTemplate.getInterestType().getId());
-
-        loanApplicationPresenter.createLoansAccount(loansPayload);
+        if (Network.isConnected(getActivity()) == true) {
+            LoansPayload loansPayload = new LoansPayload();
+            loansPayload.setClientId(loanTemplate.getClientId());
+            loansPayload.setLoanPurposeId(purposeId);
+            loansPayload.setProductId(productId);
+            loansPayload.setPrincipal(loanTemplate.getPrincipal());
+            loansPayload.setLoanTermFrequency(loanTemplate.getTermFrequency());
+            loansPayload.setLoanTermFrequencyType(loanTemplate.getInterestRateFrequencyType()
+                    .getId());
+            loansPayload.setLoanType("individual");
+            loansPayload.setNumberOfRepayments(loanTemplate.getNumberOfRepayments());
+            loansPayload.setRepaymentEvery(loanTemplate.getRepaymentEvery());
+            loansPayload.setRepaymentFrequencyType(loanTemplate.getInterestRateFrequencyType()
+                    .getId());
+            loansPayload.setInterestRatePerPeriod(loanTemplate.getInterestRatePerPeriod());
+            loansPayload.setExpectedDisbursementDate(disbursementDate);
+            loansPayload.setSubmittedOnDate(submittedDate);
+
+            loansPayload.setTransactionProcessingStrategyId(
+                    loanTemplate.getTransactionProcessingStrategyId());
+            loansPayload.setAmortizationType(loanTemplate.getAmortizationType().getId());
+            loansPayload.setTransactionProcessingStrategyId(
+                    loanTemplate.getTransactionProcessingStrategyId());
+            loansPayload.setInterestCalculationPeriodType(
+                    loanTemplate.getInterestCalculationPeriodType().getId());
+            loansPayload.setInterestType(loanTemplate.getInterestType().getId());
+
+            loanApplicationPresenter.createLoansAccount(loansPayload);
+
+        } else {
+            showError("failed to load template");
```

Comment:
This feature needs more improvement but for now change SnackBar message to strings.xml string's `no_internet_connection`